### ISSUE TITLE
Allow to reveal an empty TreeView programmatically

### ISSUE THREAD

**Issue by apupier** [microsoft/vscode#90005]
Currently, it i possible to reveal/expand a TreeView only when there is an element in the TreeView. it is oputting revealing this element in the TreeView.

it would useful to be abel to expand the TreeView even when there are no elements in the Tree.

Use case:
- when testing https://github.com/camel-tooling/vscode-camelk/pull/291
- when writing a tutorial guiding users https://github.com/redhat-developer/vscode-didact/issues/4#issuecomment-557648413

asked on StackOverflow https://stackoverflow.com/questions/59983711/how-to-programmatically-make-a-treeview-visible

**Comment by alexr00** [comment]
Proposal is to allow the first element here to be undefined:

https://github.com/microsoft/vscode/blob/e95c40c1e461f83f5ff1c68d61253ab23c85f86b/src/vs/vscode.proposed.d.ts#L1042-L1044

**Comment by eamodio** [comment]
I think we should add a `show` method, rather than overloading `reveal`. This also aligns with `WebviewViews` which have:
```ts
show(preserveFocus?: boolean): void;
```

I also think we should add a `hide` method to both tree views and webview views (which has been requested). Which also better aligns with a `show` method as we use that on lots of other API objects.

**Comment by alexr00** [comment]
I think that could result in this pattern:
```typescript
let element: T | undefined;
// ... element gets set to something
if (element) {
    tree.reveal(element);
} else {
    tree.show();
}
```

**Comment by eamodio** [comment]
It could, but imo that still feel more right to me. Also the `reveal` method has an options bag that while there is some overlap, feels strange to apply to the view itself. And I feel the alignment with `WebviewViews` is quite important as well.

**Comment by wandyezj** [comment]
I'd prefer to simply allow the reveal element to be undefined.

Alternative is to allow an overload to reveal that moves the element to the options parameter.

```typescript
 export interface TreeView<T> extends Disposable { 
 	reveal(options?: { element?: T, select?: boolean, focus?: boolean, expand?: boolean | number }): Thenable<void>; 
 } 

```

I think a `show` / `hide` API would be better as a separate proposal.

**Comment by cweijan** [comment]
I tried to pass a null to reveal, it can switch the treeview normally, which has met my needs.
```
treeView.reveal(null)
```

**Comment by gar1t** [comment]
IMHO the overload of `reveal` is objectively wrong :) `show()` is established as the function name for this (if I read the OP correctly). It's used by most (if not all) of the other view-like components (webviews, terminal, channels, etc.)

To reveal an item inside a view is view specific, whereas to make the view itself visible is something that applies to any view at all. The fact that `show()` doesn't exist on the tree view, and does pervasively elsewhere, looks like an implementation oversight.

The options in `reveal()` show that `reveal(null)` is nonsensical. These are node specific.

**Comment by gar1t** [comment]
As a driver for this behavior, I want to implement a command that shows a custom tree view in the same way that any of the other views are shown. You can do this manually via View -> Open view... -> <custom view>. This is the behavior that `tree.show()` should have IMO.

**Comment by gar1t** [comment]
Lol, `reveal(undefined)` already seems to work as described above! (I'm running 1.68.1) It's not yet documented though :)

I propose that an alias of `show()` be added (I'm glad the behavior's at least now supported!)

**Comment by EhabY** [comment]
Indeed, a dedicated `show()` would be much better especially when using a tree view with type `TreeView<SomeNode>` then we cannot pass `undefined`/`null`. I would have to (falsely) make it `TreeView<SomeNode | undefined>` and deal with the non-existent undefined or hack Typescript and do `treeView.reveal(undefined as unknown as SomeNode)`. Both options are not ideal..
